api-server/src/bin/station.rs

```diff
@@ -217,7 +233,13 @@ fn delete_routine(opts: Delete) {
                 .collect::<Vec<_>>();
             let selection = Select::with_theme(&ColorfulTheme::default())
                 .with_prompt("Which station to delete?")
-                .default(choices.len() - 1)
+                .default(match choices.len() {
```

Comment:
Kind of an odd place to check the station count. If you do this sanity check/conditional exit above the Dialoguer code then you may also be able to DRY it up easier later.